doc/tutorials/core/univ_intrin/univ_intrin.markdown

```diff
@@ -12,20 +12,177 @@ Universal Intrinsics {#tutorial_univ_intrin}
+The goal of this tutorial is to provide a guide to using the Universal Intrinsics module to vectorize your C++ code for a faster runtime.
+We'll briefly look into _SIMD intrinsics_ and how to work with wide _registers_, followed by a tutorial on the basic operations using wide registers.
+The tutorial will only demonstrate basic operations. To know more about Universal Intrinsics, visit the [documentation](https://docs.opencv.org/4.5.3/df/d91/group__core__hal__intrin.html).
 
 Theory
 ------
 
-We'll briefly look into a few concepts to better help understand the functionality. 
+In this section, we will briefly look into a few concepts to better help understand the functionality.
+
+### SIMD
+
+SIMD stands for  **S**ingle **I**nstruction, **M**ultiple **D**ata
+
+Most of the popular C++ compilers include **intrinsic** functions which allow efficient vector operations using SIMD architecture. Intrinsics are basically built in functions that are handled by the compiler.
+
+Intrinsics are often highly optimized. They, however, depend on the processor capabilities. Thus, different systems may be compatible with different “instruction sets". Popular instruction sets include **SSE, SSE2, AVX, AVX2, AVX512**, etc.
+
+If your computer supports a particular instruction set, it allows you to use intrinsics pertaining to those instructions. Systems may be compatible with more than one instruction set.
+
 
 ### Intrinsics
-Intrinsics are functions which are separately handled by the compiler. As a result, these functions are often optimized to perform in the most efficient ways possible and hence run faster than normal implementations. However, since these functions depend on the compiler, it makes it difficult to write cross-platform applications. 
+Intrinsics are functions which are separately handled by the compiler. These functions are often optimized to perform in the most efficient ways possible and hence run faster than normal implementations. However, since these functions depend on the compiler, it makes it difficult to write universal applications.
 
 ### SIMD
-SIMD stands for **S**ingle **I**nstruction, **M**ultiple **D**ata. Compilers provide SIMD Intrinsics which allow the processor to perform a single operation on a set of data simultaneously. The data is stored in what are known as *registers*. A register may be 128bits, 256bit or 512bits wide. Depending on what *Instruction Sets* your CPU supports, you may be able to use the different registers. To learn more, look [here](https:)
+SIMD stands for **S**ingle **I**nstruction, **M**ultiple **D**ata. SIMD Intrinsics allow the processor to vectorize calculations. The data is stored in what are known as *registers*. A *register* may be *128-bits*, *256-bits* or *512-bits* wide. Each *register* stores **multiple values** of the **same data type**. The size of the register and the size of each value determines the number of values stored in total.
```

Comment:
This section should be either combined with above one or named in some other way. Also please remove SIMD explanation if the section be retained separate.